File: client/FinanceTrackerClient/src/app/models/category.ts
export interface Category {
  id: string;
  name: string;
  icon: string;
  color: string;
  isDefault: boolean;
}

export interface CreateCategoryRequest {
  name: string;
  icon: string;
  color: string;
}

export interface BulkCategorizeRequest {
  transactionIds: string[];
  categoryId: string;
}

File: server/FinanceTracker.Api/Models/Category.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace FinanceTracker.Api.Models
{
    public class Category
    {
        public Guid Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Icon { get; set; } = "📁"; 
        public string Color { get; set; } = "#cccccc";
        public bool IsDefault { get; set; }
        
        public string? UserId { get; set; }
        public ApplicationUser? User { get; set; }

        public ICollection<Transaction> Transactions { get; set; } = new List<Transaction>();
    }
}